### DISCUSSION TITLE
Altering foreign key constraints

### DISCUSSION THREAD

**Discussion by adar2378** [simolus3/drift#2518] [Q&A]
Hi I want to change onDelete constraints to cascade for a table in migration. Is there any helper method to do that in drift?

**Comment by simolus3** [comment]
There is no direct way to do this in SQL, the approach that sqlite3 recommends for those migrations is to essentially re-create the table and copy data over. Drift has a API that reliably does this for you via `Migrator.alterTable`. When you don't change the type of any column (as in your case where you're only changing constraints), you can [just use `m.alterTable(TableMigration(yourTable))`](https://drift.simonbinder.eu/docs/advanced-features/migrations/#changing-column-constraints).

**Comment by adar2378** [comment]
Thank you for the response. Will try it
